picobot/painter.py

```diff
@@ -0,0 +1,224 @@
+from PIL import Image, ImageDraw, ImageFont
+from .config import ROOT_DIR
+
+# for testing purpose only
+# ROOT_DIR = '/home/diogo/Documents/CCO/Python/pico-bot/picobot'
+
+IMG_DIR = ROOT_DIR + '/images/'
+IMG_NAME = 'img'
+AVATAR_MASK_NAME = 'avatar_mask.png'
+
+AVATAR_SIZE = 42
+MARGIN = 4
+PADDING = 10
+MSG_PADDING_H = 8
+MSG_PADDING_V = 6
+TIME_PADDING_H = 8
+TIME_PADDING_V = 6
+BOX_HEIGHT = 54
+BOX_MIN_WIDTH = 160
+BOX_MAX_WIDTH = 264
+BOX_RADIUS = 10
+BOX_COLOR = "#182533"
+FONT_SIZE = 14
+TITLE_COLOR = "#338cf3"
+TEXT_COLOR = "#dddddd"
+TIME_COLOR = "#6A7B8C"
+LINE_SPACE = 10
+
+
+def draw_balloon(
+    img_draw: ImageDraw.Draw, xy: list, fill=None, width=0
+):
+    r_x0 = xy[0]
+    r_y0 = xy[1] + BOX_RADIUS
+    r_x1 = xy[2]
+    r_y1 = xy[3] - BOX_RADIUS
+    img_draw.rectangle(
+        [r_x0, r_y0, r_x1, r_y1], fill=fill, width=width
+    )
+    r_x0 = xy[0] + BOX_RADIUS
+    r_y0 = xy[1]
+    r_x1 = xy[2] - BOX_RADIUS
+    r_y1 = xy[3]
+    img_draw.rectangle(
+        [r_x0, r_y0, r_x1, r_y1], fill=fill, width=width
+    )
+    diam = 2 * BOX_RADIUS
+    c_x0 = xy[0]
+    c_y0 = xy[1]
+    c_x1 = c_x0 + diam
+    c_y1 = c_y0 + diam
+    img_draw.ellipse([c_x0, c_y0, c_x1, c_y1], fill=fill, width=width)
+    c_x0 = xy[2] - diam
+    c_x1 = c_x0 + diam
+    img_draw.ellipse([c_x0, c_y0, c_x1, c_y1], fill=fill, width=width)
+    c_y0 = xy[3] - diam
+    c_y1 = c_y0 + diam
+    img_draw.ellipse([c_x0, c_y0, c_x1, c_y1], fill=fill, width=width)
+
+    arrow_x = 10
+    arrow_y = 10 if BOX_RADIUS < 10 else BOX_RADIUS
+    x0, y0, x1, y1 = xy
+    arrow = [
+        (x0 - arrow_x, y1),
+        (x0, y1 - arrow_y),
+        (x0 + diam, y1 - arrow_y),
+        (x0 + diam, y1),
+    ]
+    img_draw.polygon(arrow, fill=fill)
+
+
+def draw_username(
+    txt_draw: ImageDraw.ImageDraw,
+    xy: list,
+    username="Jojo",
+    fill=TITLE_COLOR,
+    font=None,
+):
+    x0 = xy[0] + MSG_PADDING_H
+    y0 = xy[1] + MSG_PADDING_V
+    txt_draw.text((x0, y0), username, font=font, fill=fill)
+
+
+def draw_message(
+    txt_draw: ImageDraw.ImageDraw,
+    xy: list,
+    text="Oi, eu sou o Jojo",
+    fill=TEXT_COLOR,
+    font=None,
+):
+    tw = txt_draw.textsize(text, font=font)
+    x0 = xy[0] + MSG_PADDING_H
+    y0 = xy[3] - (MSG_PADDING_V + tw[1])
+    txt_draw.text((x0, y0), text, font=font, fill=fill)
+
+
+def draw_time(
+    txt_draw: ImageDraw.ImageDraw, xy: list, text="04:20", fill=TEXT_COLOR, font=None
+):
+    tw = txt_draw.textsize(text, font=font)
+    x0 = xy[2] - TIME_PADDING_H - tw[0]
+    y0 = xy[3] - (TIME_PADDING_V + tw[1])
+    txt_draw.text((x0, y0), text, font=font, fill=TIME_COLOR)
+
+
+def draw_avatar(img, avatar_path: str):
+    y0 = BOX_HEIGHT + PADDING - AVATAR_SIZE
+    y1 = BOX_HEIGHT + PADDING
+    xy = [MARGIN, y0, MARGIN + AVATAR_SIZE, y1]
+    box = tuple(a-2 for a in xy[0:2])
+    if avatar_path == '':
```

Comment:
I think this check should be on the caller:

```python
    if avatar_path:
        draw_avatar(img, avatar_path)
```

`draw_avatar` should presume `avatar_path` has a valid path.

Also, optional arguments should be `Optional[<type>]`